### ISSUE TITLE
JUnit formatter improperly reporting errors and warning

### ISSUE THREAD

**Issue by dinubs** [eslint/eslint#11326]
**Tell us about your environment**

* **ESLint Version:** 5.3.0
* **Node Version:** 11.1.0
* **npm Version:** 6.4.1

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
    "env": {
        "browser": true,
        "es6": true
    },
    "extends": "eslint:recommended",
    "parserOptions": {
        "ecmaVersion": 2015,
        "sourceType": "module"
    },
    "rules": {
        "no-console": "warn"
    }
}

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
function cube(x) {
  console.log('ok')
  return x * 3;
}

cube(3);
```

```bash
$(yarn bin)/eslint -f junit index.js > eslint-result.xml
```

**What did you expect to happen?**

Expected formatter to follow the JUnit [spec](https://www.ibm.com/support/knowledgecenter/en/SSUFAU_1.0.0/com.ibm.rsar.analysis.codereview.cobol.doc/topics/cac_useresults_junit.html). See the analysis result section, it should have a type field specifying the severity of the error, rather than changing the tag of the xml like it is currently doing.

Reason why this is important is that in CI systems such as jenkins, they look at the type field to determine whether to show it in the "tests" tab. On large codebases there could be a lot of warnings, and 0 errors but in the current state of eslint the JUnit formatter will output warnings and errors. 

There is a workaround to this currently, but it's less than ideal, you can run eslint with the `--quiet` command and will then only report errors.

**What actually happened? Please include the actual, raw output from ESLint.**

```
<?xml version="1.0" encoding="utf-8"?>
<testsuites>
<testsuite package="org.eslint" time="0" tests="1" errors="1" name="/Users/gavin/Documents/test/index.js">
<testcase time="0" name="org.eslint.no-console"><failure message="Unexpected console statement."><![CDATA[line 2, col 3, Warning - Unexpected console statement. (no-console)]]></failure></testcase>
</testsuite>
</testsuites>
```

**Comment by Exelord** [comment]
Can we reopen that issue? Its still a pain :/ Warning shouldn't be reported in junit format as there is no support for them.
